jwst/cube_build/blot_cube_build.py

```diff
@@ -131,98 +131,144 @@ def blot_images(self):
+            wave_blot = lam_det[good_data]
                 
             crval1 = model.meta.wcsinfo.crval1
             crval2 = model.meta.wcsinfo.crval2
-            xi_blot,eta_blot = coord.radec2std(crval1, crval2,ra_blot,dec_blot) # for each x,y pixel
 
-            xi_cube,eta_cube = coord.radec2std(crval1, crval2, # cube values
-                                                   self.cube_ra,self.cube_dec) 
+            xi_blot,eta_blot = coord.radec2std(crval1, crval2,
+                                               ra_blot,dec_blot) 
 
+            xi_cube,eta_cube = coord.radec2std(crval1, crval2, 
+                                               self.cube_ra,self.cube_dec) 
+            
             nplane = self.naxis1 * self.naxis2
             self.xi_centers = np.reshape(xi_cube[0,:,:],nplane)
             self.eta_centers =np.reshape(eta_cube[0,:,:],nplane)
-                
+
             num = ra_blot.size
-#                print('Size of pixel detectors looping over',num)
-            iprint = 0 
+            iprint = 0
+
 #________________________________________________________________________________  
-               # loop over the valid pixels on the detector
+#loop over the median cube pixels that fall within the ROI
+# of the detector center
+            ii = 0 
             for ipt in range(0, num - 1):
+#                if ii == print_limit: 
+#                    log.info('On point %i out of %i',ipt,num)
+#                ii = ii +1 
+#                if ii > print_limit: ii = 0
                 # xx,yy are the index value of the orginal detector frame -
                 # blot image
                 yy = y[ipt]
-                xx = x[ipt]                    
-                    # find the cube values that fall withing ROI of detector xx,yy
+                xx = x[ipt]  
+                
+                # find the cube values that fall withing ROI of detector xx,yy
                 xdistance = (xi_blot[ipt] - self.xi_centers)
                 ydistance = (eta_blot[ipt] -self.eta_centers)
-
                 radius = np.sqrt(xdistance * xdistance + ydistance * ydistance)
-                
                 indexr = np.where(radius  <=self.rois)
-                indexz = np.where(abs(self.lam_centers - wave_blot[ipt]) <= self.roiw)
+                indexz = np.where(abs(self.lam_centers- wave_blot[ipt]) <= self.roiw)
+                zdistance = wave_blot[ipt] - self.lam_centers
 
-                wave_found = self.lam_centers[indexz]        # z Cube values falling in wavelength roi
-                xi_found = self.xi_centers[indexr]   # x Cube values within radius 
-                eta_found = self.eta_centers[indexr]  # y cube values with the radius
+                # Find the Cube spaxels falling with ROI regions
+                wave_found = self.lam_centers[indexz] 
+                xi_found = self.xi_centers[indexr]    
+                eta_found = self.eta_centers[indexr]  
```

Comment:
I lost the logic here because I'm not familiar with the algorithm but otherwise it looks good.